test/support/usage_example_for_dialyzer.ex

```diff
@@ -0,0 +1,10 @@
+defmodule Example do
```

Comment:
Is this really necessary? I feel it weird to have a module just to check that.
I mean what is the difference between that and call it in a test

Reply:
As I understand it from a co-worker, dialyzer checks compiled code and as such usage in `.exs` tests are not sufficient to cause dialyzer to flag the erroneous typespec.

Reply:
I see but we don't expect test phase to check dialyzer. Tests must focus on checking a behaviour instead of types. 

However you come with an interesting problem. Let's see how we could find a good way of achieving that.

For a fast merge, could you remove this file and open an issue to introduce the problem?

Reply:
Addressed